### DISCUSSION TITLE
Runtime error Connection timed out

### DISCUSSION THREAD

**Discussion by shassan-el** [gijzelaerr/python-snap7#415] [Q&A] [answered]
When I try running the client.connect line below, I get a runtime error:

```
import snap7
client = snap7.client.Client()
client.connect(IP_ADDRESS, RACK, SLOT)
```

Runtime error:
`RuntimeError: b' TCP : Connection timed out'`

Any ideas what causing this and how I can fix it?

**Comment by swamper123** [comment]
Can you ping your device with `IP_ADDRESS`?
Have you made the configs on your device?

**Reply by shassan-el** [reply]
Yes I can ping the IP address from my computer successfully with 0 packets lost.
What configs do I need to make? The only thing I have done is download the DLL file and save it in virtual environment directory which has worked (previously it couldn't find the DLL when I placed in Windows/Systems32 folder)

**Reply by shassan-el** [reply]
I have now managed to connect, but I had to wire my laptop directly into the Siemens Scalence ethernet switch (which is connect to the PLC) by ethernet and then set the subnet to the same as PLC. Any idea how I can remotely connect to the PLC? Would I need some sort of gateway connected to the PLC with an IP address that's accessible from the on-site wireless network?

**Comment by gijzelaerr** [comment, marked as answer]
Closing as stale. TCP connection timeout is a network connectivity issue.
